source/simulator/assembly.cc

```diff
@@ -375,21 +375,25 @@ namespace aspect
     if (parameters.include_melt_transport)
       stokes_dofs_per_cell += finite_element.base_element(introspection.variable("compaction pressure").base_index).dofs_per_cell;
 
+    auto worker = [&](const typename DoFHandler<dim>::active_cell_iterator &cell,
+                      internal::Assembly::Scratch::StokesPreconditioner<dim> &scratch,
+                      internal::Assembly::CopyData::StokesPreconditioner<dim> &data)
+    {
+      this->local_assemble_stokes_preconditioner(cell, scratch, data);
+    };
+
+    auto copier = [&](const internal::Assembly::CopyData::StokesPreconditioner<dim> &data)
+    {
+      this->copy_local_to_global_stokes_preconditioner(data);
+    };
+
     WorkStream::
     run (CellFilter (IteratorFilters::LocallyOwnedCell(),
                      dof_handler.begin_active()),
          CellFilter (IteratorFilters::LocallyOwnedCell(),
                      dof_handler.end()),
-         std::bind (&Simulator<dim>::
-                    local_assemble_stokes_preconditioner,
-                    this,
-                    std::placeholders::_1,
-                    std::placeholders::_2,
-                    std::placeholders::_3),
-         std::bind (&Simulator<dim>::
-                    copy_local_to_global_stokes_preconditioner,
-                    this,
-                    std::placeholders::_1),
+         worker,
+         copier,
```

Comment:
I like this, the call was way to complex before. I would prefer `work_function` and `copy_function`, but if you prefer as is, it is not a big deal.

Reply:
Agree. I don't care much about the name, but this mirrors the argument names in the ``WorkStream::run`` function.